### PR TITLE
IBP-3459 API Permissions

### PR DESCRIPTION
**Pull request by mderamos**
Added PreAuthorize Annotations

### CODE REVIEW

File: src/main/java/org/ibp/api/rest/program/ProgramResource.java [hunk 1]
```diff
@@ -14,14 +16,14 @@
...
-@RequestMapping("/program")
+@PreAuthorize("hasAnyAuthority('ADMIN','ADMINISTRATION','CROP_MANAGEMENT','SITE_ADMIN')")
 public class ProgramResource {
 
     @Autowired
     private ProgramService programService;
 
-    @RequestMapping(method = RequestMethod.GET)
-    public ResponseEntity<List<ProgramSummary>> listPrograms(@RequestParam final String cropName) {
+    @RequestMapping(value = "/crops/{cropName}/program", method = RequestMethod.GET)
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/rest/program/ProgramResource.java:25]: Should be /crops/{cropName}/program**s** to follow REST standard

File: src/main/java/org/ibp/api/rest/program/ProgramResource.java [hunk 2]
```diff
@@ -14,14 +16,14 @@
...
     @Autowired
     private ProgramService programService;
 
-    @RequestMapping(method = RequestMethod.GET)
-    public ResponseEntity<List<ProgramSummary>> listPrograms(@RequestParam final String cropName) {
+    @RequestMapping(value = "/crops/{cropName}/programs", method = RequestMethod.GET)
+    public ResponseEntity<List<ProgramSummary>> listPrograms(@PathVariable final String cropName) {
         return new ResponseEntity<>(this.programService.listProgramsByCropName(cropName), HttpStatus.OK);
     }
-}
```
Comment (nahuel-soldevilla) [inline src/main/java/org/ibp/api/rest/program/ProgramResource.java:27]: @mderamos @darla-leafnode 

Can we keep both and mark `/program` as `@Deprecated`? For backwards compatibility, maybe there's someone out there using this method :disappointed: 

cc @clarysabel @mcrimi

Comment (darla-leafnode) [reply src/main/java/org/ibp/api/rest/program/ProgramResource.java:27]: @nahuel-soldevilla - i don't think there's a need to keep it as it's an internal API. It has a counterpart brapi call which is likely the one used: `GET​/{crop}​/brapi​/v1​/programs`

As an aside, in the API permissions work ([IBP-3039](https://ibplatform.atlassian.net/browse/IBP-3039)) for internal services, there were quite a handful of internal services that were renamed to be able to work for crop and program-level permissions checking..
For example all the ontology resources in the old format of "/ontology/{cropname}/xxx" were renamed to "crops/{cropname}/xxx". It's all documented [here](https://ibplatform.atlassian.net/wiki/spaces/DEV/pages/832012689/Implementing+Permissions+in+BMSAPI) so this is actually just a leftover from that analysis and implementation.

Comment (nahuel-soldevilla) [reply src/main/java/org/ibp/api/rest/program/ProgramResource.java:27]: @darla-leafnode 
Makes sense, and I guess it's not a big deal also, in this case we are just renaming, so if someone is using it they can just adjust accordingly.

**Pull request merged**